### PR TITLE
Added Multi-partition read #106

### PR DESCRIPTION
**Pull request by alkampfergit**
Added multi partition read to basic persistence.

### CODE REVIEW

File: src/NStore.BaseSqlPersistence/BaseSqlPersistenceOptions.cs [hunk 1]
```diff
@@ -34,14 +36,21 @@ protected BaseSqlPersistenceOptions(INStoreLoggerFactory loggerFactory)
...
         public abstract string GetRangeSelectChunksSql(
             long upperIndexInclusive,
             long lowerIndexInclusive,
             int limit,
             bool descending
         );
 
+        public abstract string GetRangeMultiplePartitionSelectChunksSql(
+            IEnumerable<string> partitionIdsList,
+            long upperIndexInclusive,
```
Comment (AGiorgetti) [inline src/NStore.BaseSqlPersistence/BaseSqlPersistenceOptions.cs:49]: ordine dei parametri controintuitivo

**Review by AGiorgetti — APPROVED** [review]

File: src/NStore.Core/Persistence/IPartitionPersistenceExtended.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        /// (Partition_X, 1), (Partition_Y, 1), (Partition_Z, 1), (Partition_Y, 2), (Partition_Y, 3), (Partition_X, 2)
+        /// </para>
+        /// <para>
+        /// If we search for X and Y we can have both sequence
+        /// (Partition_X, 1), (Partition_X, 2), (Partition_Y, 1), (Partition_Y, 2), (Partition_Y, 3)
+        /// (Partition_X, 1), (Partition_Y, 1), (Partition_Y, 2), (Partition_Y, 3), (Partition_X, 2)
+        /// </para>
+        /// <para>
+        /// The only guarantee is that we cannot have for the SAME partition an unordered sequence of version but
+        /// no ordering is guaranteed between partition, even if you perform subsequent call, each call can
```
[the remark this answers is not in the document]
Comment (alkampfergit) [reply src/NStore.Core/Persistence/IPartitionPersistenceExtended.cs:41]: Bene in una frase ne ho sbagliati 3, potevo fare peggio :D

File: src/NStore.BaseSqlPersistence/BaseSqlPersistenceOptions.cs (hunk 1 above)
Comment (alkampfergit) [reply src/NStore.BaseSqlPersistence/BaseSqlPersistenceOptions.cs:49]: I know, but here we have the other version so I kept ordering, I'l refactor both call.
image

**Comment by alkampfergit** [comment]
Approved by @AGiorgetti merge in develop and wait for @andreabalducci to close on master with a new release.

**Pull request merged**